py-polars/polars/functions/range.py

```diff
@@ -343,13 +342,13 @@ def date_range(
             )
 
         if time_zone is not None and start.tzinfo is not None:
-            if _tzinfo_to_str(start.tzinfo) != time_zone:
+            if str(start.tzinfo) != time_zone:
```

Comment:
now that only `timezone.available_timezones` are allowed as time zones, `_tzinfo_to_str` is no longer needed